src/bufreader.rs

```diff
@@ -13,6 +13,7 @@ use std::io;
 use std::io::prelude::*;
 use std::mem;
 
+#[derive(Debug)]
```

Comment:
Could this type actually have a custom debug impl like `BufReader` in the standard library? That way we can avoid accidentally dumping out kilobytes of raw data :)